### PR TITLE
Terra 0.7 update

### PR DESCRIPTION
**Pull request by gadial**
This pull request brings JKU up to date with the latest Terra stable.

### CODE REVIEW

File: qiskit_jku_provider/qasm_simulator_jku.py [hunk 1]
```diff
@@ -23,13 +23,16 @@
 from collections import OrderedDict, Counter
 import numpy as np
 
-from qiskit.backends import BaseBackend
+#from qiskit.backends import BaseBackend
 from qiskit.backends.aer import AerJob
```
Comment (diego-plan9) [inline qiskit_jku_provider/qasm_simulator_jku.py]: Can you replace this import, in order not to depend on `aer`? Ideally using your own `Job` class (similar to https://github.com/Qiskit/qiskit-projectq-provider/blob/master/qiskit_addon_projectq/projectqjob.py).

File: qiskit_jku_provider/qasm_simulator_jku.py [hunk 2]
```diff
@@ -23,13 +23,16 @@
...
 from qiskit.backends.aer._simulatorerror import SimulatorError
```
Comment (diego-plan9) [inline qiskit_jku_provider/qasm_simulator_jku.py]: Also, can you replace this import with your own error class, inheriting from `QiskitError`, as in https://github.com/Qiskit/qiskit-projectq-provider/blob/master/qiskit_addon_projectq/projectqsimulatorerror.py ?

File: qiskit_jku_provider/qasm_simulator_jku.py [hunk 3]
```diff
@@ -23,13 +23,16 @@
 from collections import OrderedDict, Counter
 import numpy as np
 
-from qiskit.backends import BaseBackend
+#from qiskit.backends import BaseBackend
```
Comment (diego-plan9) [inline qiskit_jku_provider/qasm_simulator_jku.py]: Can you make a round of removing commented-out lines similar to this one, for readability? Usually we can just refer to git history to see what lines have changed if needed!

File: setup.py [hunk 4]
```diff
@@ -65,13 +91,13 @@ def has_ext_modules(self):
         return True
     
 setup(
-    name="qiskit-addon-jku",
+    name="qiskit-jku-provider",
     version="1.0.2",
```
[the remark this answers is not in the document]
Comment (gadial) [reply setup.py]: Updated to 1.1.0

Will remove the old pypi repo after uploading the new one.

File: test_jku/test_qasm_simulator_jku.py [hunk 5]
```diff
@@ -18,8 +18,8 @@
 from qiskit import QuantumCircuit
 from qiskit import QuantumRegister
 from qiskit import ClassicalRegister
-from qiskit.wrapper import get_backend
-from qiskit_addon_jku import QasmSimulatorJKU
+from qiskit import Aer
```
[the remark this answers is not in the document]
Comment (gadial) [reply test_jku/test_qasm_simulator_jku.py]: Thanks! I was not aware of BasicAer. Done.

File: qiskit_jku_provider/qasm_simulator_jku.py (hunk 3 above)
Comment (gadial) [reply qiskit_jku_provider/qasm_simulator_jku.py]: You are correct, of course. Done.

File: qiskit_jku_provider/qasm_simulator_jku.py (hunk 1 above)
Comment (gadial) [reply qiskit_jku_provider/qasm_simulator_jku.py]: Done, thanks.

File: qiskit_jku_provider/qasm_simulator_jku.py (hunk 2 above)
Comment (gadial) [reply qiskit_jku_provider/qasm_simulator_jku.py]: Sure thing, done.

**Pull request merged**